frame/fast-unstake/src/lib.rs

```diff
@@ -254,7 +267,17 @@ pub mod pallet {
 				Head::<T>::get().map_or(true, |UnstakeRequest { stash, .. }| stash != stash),
 				Error::<T>::AlreadyHead
 			);
-			Queue::<T>::remove(stash);
+			let deposit = Queue::<T>::take(stash.clone());
+
+			if let Some(deposit) = deposit.defensive() {
+				let remaining = T::DepositCurrency::unreserve(&stash, deposit);
+				if !remaining.is_zero() {
+					frame_support::defensive!("`not enough balance to unreserve`");
+					ErasToCheckPerBlock::<T>::put(0);
+					Self::deposit_event(Event::<T>::InternalError)
```

Comment:
This then goes on to be Ok(()) - should it be an error?

Reply:
That's alright, it has defensive checks, what this basically means is that this should always be a defensive path and if it happens - that'd panic.

Reply:
And in prod scenario this simply disables the pallet until further notice.

Reply:
yes -- a defensive code path is one that should never happen. We're just coding what should happen in case there's a bug. Returning Ok(()), and 

```
ErasToCheckPerBlock::<T>::put(0);
Self::deposit_event(Event::<T>::InternalError)
```

is as good as it gets.